src/application/worker/index.worker.js

```diff
@@ -9,42 +9,105 @@ async function start() {
   const store = require("./store").default;
   const loop = require("./loop").default;
   const grabCanvasPlugin = require("../plugins/grab-canvas").default;
+  const get = require("lodash.get");
 
   const { tick: frameTick } = require("./frame-counter");
   const { getFeatures, setFeatures } = require("./audio-features");
   // const featureAssignmentPlugin = require("../plugins/feature-assignment");
 
   let interval = store.getters["fps/interval"];
 
+  const commitQueue = [];
+
   store.subscribe(mutation => {
-    const { type, payload } = mutation;
+    const { type: mutationType, payload: mutationPayload } = mutation;
 
-    if (type === "beats/SET_BPM" || type === "fps/SET_FPS") {
-      store.dispatch("tweens/updateBpm", { bpm: payload.bpm });
+    if (mutationType === "beats/SET_BPM" || mutationType === "fps/SET_FPS") {
+      store.dispatch("tweens/updateBpm", { bpm: mutationPayload.bpm });
     }
 
-    if (type === "beats/SET_KICK" && payload.kick === lastKick) {
+    if (
+      mutationType === "beats/SET_KICK" &&
+      mutationPayload.kick === lastKick
+    ) {
       return;
-    } else if (type === "beats/SET_KICK" && payload.kick !== lastKick) {
-      lastKick = payload.kick;
+    } else if (
+      mutationType === "beats/SET_KICK" &&
+      mutationPayload.kick !== lastKick
+    ) {
+      lastKick = mutationPayload.kick;
     }
 
-    if (type === "fps/SET_FPS") {
+    if (mutationType === "fps/SET_FPS") {
       interval = store.getters["fps/interval"];
     }
 
     if (
-      type === "modules/UPDATE_ACTIVE_MODULE" &&
-      (payload.key !== "props" || payload.key !== "meta")
+      mutationType === "modules/UPDATE_ACTIVE_MODULE" &&
+      (mutationPayload.key !== "props" || mutationPayload.key !== "meta")
     ) {
       return;
     }
 
+    const {
+      inputs: { inputs, inputLinks }
+    } = store.state;
+
+    // Update mutation type Input Links
```

Comment:
What is an input link of type `mutation`?

Reply:
There are multiple types of input links.
* all input links
  * Params:
    * inputId
      * String. The ID of the input to link to
    * ?source
      * String. Identifies what created and/or manages this input link
    * min
      * Number. Minimum expected value, will scale the incoming value to the input's min/max if min and max are set
    * max
      * Number. Maximum expected value, will scale the incoming value to the input's min/max if min and max are set
* getter  _*:warning: updates every frame*_
  * Params:
    * location
      * String. location of the getter in the store, e.g. `meyda/getFeature`
    * ?args
      * Array. arguments for a getter that is a function, e.g. `["energy"]`
* mutation
  * Params:
    * location
      * String. location of the value to read in the store, e.g. `midi.devices[1].channelData[1][144]`
    * match
      * Object. Parameters of the mutation to match. Has keys:
        * type
          * String. Mutation type from the store e.g. "midi/WRITE_DATA".
        * ?payload
          * Object. Key-value pairs to match in the mutation payload. e.g.
          ```JavaScript
          {
            id: `${id}-${name}-${manufacturer}`,
            channel,
            type
          } 
          ```
* state _*:warning: updates every frame*_
  * Params:
    * location
      * String. location of the value to read in the store, e.g. `midi.devices[1].channelData[1][144]`

state and getter links are updated in loop.js
https://github.com/vcync/modV/blob/322385e6a59aa075c80c240a783478f95835853e/src/application/worker/loop.js#L89

Reply:
@2xAA thanks for this explanation!